integration-tests/src/test/java/com/datastax/oss/driver/api/core/cql/ExecutionInfoWarningsIT.java

```diff
@@ -0,0 +1,191 @@
+import com.datastax.oss.driver.categories.ParallelizableTests;
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoaderBuilder;
+import com.datastax.oss.driver.internal.core.cql.CqlRequestHandler;
+import com.google.common.base.Strings;
+import java.util.List;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.slf4j.LoggerFactory;
+
+@Category(ParallelizableTests.class)
+public class ExecutionInfoWarningsIT {
+
+  private static final int MAX_QUERY_LENGTH = 500;
+  private static final CcmRule CCM = CcmRule.getInstance();
+  private static final SessionRule<CqlSession> SESSION_RULE =
+      SessionRule.builder(CCM)
+          .withConfigLoader(
+              SessionUtils.configLoaderBuilder()
+                  .withInt(DefaultDriverOption.REQUEST_PAGE_SIZE, 20)
+                  .withInt(DefaultDriverOption.REQUEST_LOGGER_MAX_QUERY_LENGTH, MAX_QUERY_LENGTH)
+                  .withProfile(
+                      "log-disabled",
+                      DefaultDriverConfigLoaderBuilder.profileBuilder()
+                          .withString(DefaultDriverOption.REQUEST_DISABLE_LOG_WARNINGS, "true")
+                          .build())
+                  .build())
+          .build();
+  private static final String KEY = "test";
+
+  @ClassRule public static TestRule ccmChain = RuleChain.outerRule(CCM).around(SESSION_RULE);
+
+  private final Appender<ILoggingEvent> appender =
+      (Appender<ILoggingEvent>) Mockito.mock(Appender.class);
+  private final ArgumentCaptor<ILoggingEvent> loggingEventCaptor =
+      ArgumentCaptor.forClass(ILoggingEvent.class);
+  private Logger logger;
+  private Level originalLoggerLevel;
+
+  @BeforeClass
+  public static void setupSchema() {
+    // table with simple primary key, single cell.
+    SESSION_RULE
+        .session()
+        .execute(
+            SimpleStatement.builder("CREATE TABLE IF NOT EXISTS test (k int primary key, v text)")
+                .withExecutionProfile(SESSION_RULE.slowProfile())
+                .build());
+    for (int i = 0; i < 100; i++) {
+      SESSION_RULE
+          .session()
+          .execute(
+              SimpleStatement.builder("INSERT INTO test (k, v) VALUES (?, ?)")
+                  .addPositionalValues(KEY, i)
+                  .build());
+    }
+  }
+
+  @Before
+  public void setupLogger() {
+    // setup the log appender
+    logger = (Logger) LoggerFactory.getLogger(CqlRequestHandler.class);
+    originalLoggerLevel = logger.getLevel();
+    logger.setLevel(Level.WARN);
+    logger.addAppender(appender);
+  }
+
+  @After
+  public void cleanupLogger() {
+    logger.setLevel(originalLoggerLevel);
```

Comment:
Will this `@Before` and `@After` work properly together with `ParallelizableTests`? Isn't there a possiblity that this will work in a non-deterministic way?

Reply:
Right, if the default level is `ERROR` and other tests also manipulate this particular level dynamically, there could be race conditions. The test should be serial (no `@Category` annotation, build the `CcmRule` with `CustomCcmRule.builder().build()`).

Now that you mention it, this could be an issue in the 1.x PR as well.

Reply:
For the 1.x PR, I have the only test that changes the System property is marked with `isolated` here: https://github.com/datastax/java-driver/pull/1178/files#diff-7a9fae4b8210f01e97727cc6bf26e1ecR117

Reply:
> The test should be serial (no `@Category` annotation, build the `CcmRule` with `CustomCcmRule.builder().build()`).

Hmm why use `CustomCmmRule` instead of `@Category`? This won't prevent the race condition. Imo we should add `@Category(IsolatedTests.class)` (but the CCM rule could be the non-custom one).

Reply:
> Though now that I look at it again, I'm not changing the log level in the 1.x PR.

But you do set a system property, so it's OK.